source/features/search-releases.tsx

```diff
@@ -0,0 +1,267 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import OptionsSync from 'webext-options-sync';
+
+import * as api from '../libs/api';
+import features from '../libs/features';
+import {getOwnerAndRepo} from '../libs/utils';
+import {isReleasesOrTags} from '../libs/page-detect';
+import {checkInline, x as closeIcon} from '../libs/icons';
+
+type ReleasesResponseV4 = {
+	releases: {
+		totalCount: number;
+		edges: [
+			{
+				node: {
+					tagName: string;
+				};
+			}
+		];
+	};
+};
+
+/* eslint-disable @typescript-eslint/camelcase */
+type ReleasesResponseV3 = {
+	tag_name: string;
+	html_url: string;
+};
+/* eslint-enable @typescript-eslint/camelcase */
+
+type Release = {
+	tag: string;
+	htmlPage: string;
+};
+
+async function init() {
+	if (!isReleasesOrTags()) {
+		console.debug('Not on releases or tag page');
+
+		return false;
+	}
+
+	if (!isReleasesTabSelected()) {
+		console.debug('Release tab is not selected.');
+
+		return false;
+	}
+
+	if (!hasAnyReleases()) {
+		console.debug('No releases present for the selected repository.');
+
+		return false;
+	}
+
+	const {personalToken} = await new OptionsSync().getAll();
+
+	const releases = await fetchReleases(personalToken);
+	addSelectMenu(releases);
+	registerEventListener();
+}
+
+const isReleasesTabSelected = () => getSelectedTabText() === 'Releases';
+
+const hasAnyReleases = () => select.exists('.release-entry') || select.exists('.release');
+
+const getSelectedTabText = () => {
+	const selectedTab = select('.subnav > .subnav-links > .selected');
+
+	return selectedTab ? selectedTab.innerText : null;
+};
+
+const fetchReleases = async (personalToken): Promise<Release[]> => {
+	const {ownerName, repoName} = getOwnerAndRepo();
+
+	if (personalToken && personalToken.trim().length > 0) {
+		return fetchReleasesV4(ownerName, repoName);
+	}
+
+	return fetchReleasesV3(ownerName, repoName);
+};
+
+const fetchReleasesV4 = async (ownerName, repoName): Promise<Release[]> => {
+	try {
+		const jsonResponse: Record<'repository', ReleasesResponseV4> = await api.v4(
+			`{
+			repository(owner: ${ownerName}, name: ${repoName}) {
+				releases(first: 100, orderBy: { field: CREATED_AT, direction: DESC }) {
+					totalCount
+					edges {
+						node {
+							tagName
+						}
+					}
+				}
+			}
+		}`
+		);
+
+		const releases: Release[] = jsonResponse.repository.releases.edges.map(
+			({node: {tagName}}) => {
+				return {
+					tag: tagName,
+					htmlPage: `https://github.com/${ownerName}/${repoName}/releases/tag/${tagName}`
+				};
+			}
+		);
+
+		return releases;
+	} catch (error) {
+		return Promise.reject(error);
+	}
+};
+
+const fetchReleasesV3 = async (ownerName, repoName): Promise<Release[]> => {
+	try {
+		const jsonResponse = await api.v3(`repos/${ownerName}/${repoName}/releases?per_page=100`) as ReleasesResponseV3[];
+
+		const releases: Release[] = jsonResponse.map(item => {
+			return {tag: item.tag_name, htmlPage: item.html_url};
+		});
+
+		return releases;
```

Comment:
If you're not going to use the variable, just return on line 118

Reply:
Code removed.